url.py

```diff
@@ -0,0 +1,71 @@
+import hashlib
+
+
+class URL:
```

Comment:
The way you've broken up the classes now is much more how I would expect them to be. Good job!